src/inmanta/ast/constraint/expression.py

```diff
@@ -331,7 +331,9 @@ class Regex(UnaryOperator):
 
     def __init__(self, op1: ExpressionStatement, op2: str):
         self.regex = re.compile(op2)
-        UnaryOperator.__init__(self, "regex", op1)
+        # This class used to inherit from BinaryOperator,
+        # We make this call to the shared super class to maintain backward compatibility
+        Operator.__init__(self, "regex", [op1, Literal(self.regex)])
```

Comment:
Another option would be to revert the changes to `Regex` and make it a `BinaryOperator` again. If we interpret it more as the `matching` operator that makes some sense, and it would get rid of this unconventional structure.

Reply:
ok